### PR TITLE
Generate documentation

### PR DESCRIPTION
**Pull request by mbaechler**
only last commit matters

### CODE REVIEW

File: server/protocols/jmap/doc/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM python:2.7.11
+
+RUN pip install mkdocs
+
+ADD . /doc
```
Comment (fvn-linagora) [inline server/protocols/jmap/doc/Dockerfile]: COPY is preferred (see:https://docs.docker.com/engine/articles/dockerfile_best-practices/#add-or-copy)

File: server/protocols/jmap/doc/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+WORKDIR /doc
+
+RUN mkdocs build --clean
+
+WORKDIR site
+
+EXPOSE 8000
+
+CMD python -m SimpleHTTPServer 8000
```
Comment (fvn-linagora) [inline server/protocols/jmap/doc/Dockerfile]: should probably not run as root (USER instruction)

File: server/protocols/jmap/doc/Dockerfile (hunk 1 above)
Comment (mbaechler) [inline server/protocols/jmap/doc/Dockerfile]: good to know

File: server/protocols/jmap/doc/Dockerfile (hunk 2 above)
Comment (mbaechler) [inline server/protocols/jmap/doc/Dockerfile]: really ? what do you expect from this image ?

**Comment by rouazana** [comment]
:+1:

**Comment by aduprat** [comment]
:+1:

**Comment by mbaechler** [comment]
@fvn-linagora : it is more complex but as you did it, I copy/paste your code in the Dockerfile so you can be happy.

**Comment by fvn-linagora** [comment]
:+1

**Comment by mbaechler** [comment]
merged

**Pull request closed without merging**